src/datapattern.jl

```diff
@@ -23,9 +23,12 @@ LearnBase.gettarget(::typeof(identity), dt::DataFrameRow) =
     _throw_table_error()
 
 # convenience syntax to allow column name
-LearnBase.gettarget(col::Symbol, dt::AbstractDataFrame) = dt[1, col]
+LearnBase.gettarget(col::Symbol, dt::AbstractDataFrame) =
+    LearnBase.gettarget(col, only(dt))
 LearnBase.gettarget(col::Symbol, dt::DataFrameRow) = dt[col]
-LearnBase.gettarget(fun, dt::AbstractDataFrame) = fun(dt)
+LearnBase.gettarget(fun, dt::AbstractDataFrame) =
```

Comment:
If I understand the intention correctly the `fun` form should also normally take a `DataFrameRow` as an argument. Right? This change is breaking (and I might have misunderstood the intention here).